substrate/frame/multisig/src/migrations/v2/mod.rs

```diff
@@ -0,0 +1,155 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::PALLET_MIGRATIONS_ID;
+use crate::{pallet::Config, Multisig, Multisigs};
+use frame_support::{
+	migrations::{SteppedMigration, SteppedMigrationError},
+	pallet_prelude::PhantomData,
+	weights::WeightMeter,
+};
+
+mod benchmarks;
+mod tests;
+pub mod weights;
+
+/// Module containing the OLD (v1) storage items.
+///
+/// Before running this migration, the storage alias defined here represents the
+/// `on_chain` storage.
+mod v1 {
+	use crate::pallet::Config;
+	use frame_support::pallet_prelude::*;
+	use frame_support::storage_alias;
+
+	use crate::{BalanceOf, BlockNumberFor, Timepoint};
+
+	/// An open multisig operation.
+	#[derive(Decode)]
+	pub struct OldMultisig<BlockNumber, Balance, AccountId, MaxApprovals>
+	where
+		MaxApprovals: Get<u32>,
+	{
+		/// The extrinsic when the multisig operation was opened.
+		pub when: Timepoint<BlockNumber>,
+		/// The amount held in reserve of the `depositor`, to be returned once the operation ends.
+		pub deposit: Balance,
+		/// The account who opened it (i.e. the first to approve it).
+		pub depositor: AccountId,
+		/// The approvals achieved so far, including the depositor. Always sorted.
+		pub approvals: BoundedVec<AccountId, MaxApprovals>,
+	}
+
+	#[storage_alias]
+	pub type Multisigs<T: Config> = StorageDoubleMap<
+		_, // TODO: empty prefix is not allowed, what value should be supplied here?
+		Twox64Concat,
+		T::AccountId,
+		Blake2_128Concat,
+		[u8; 32],
+		OldMultisig<BlockNumberFor<T>, BalanceOf<T>, T::AccountId, T::MaxSignatories>,
+	>;
+}
+
+use crate::MaxEncodedLen;
+use crate::{Decode, Encode};
+// TODO: did not want to touch frame/support/src/migrations.rs
```

Comment:
in the examples, `MigrationId` was there in `frame/support/src/migrations.rs`. I didn't modify that file for granular and small-scoped PR.